admin/spec/back-end/service/pupil-access-arrangements.service.spec.js

```diff
@@ -159,4 +160,25 @@ describe('pupilAccessArrangementsService', () => {
       )
     })
   })
+  describe('deletePupilAccessArrangements', () => {
+    it('returns pupil data when successfully deleting relevant access arrangements', async () => {
+      spyOn(pupilDataService, 'sqlFindOneBySlugAndSchool').and.returnValue({id: 1, foreName: 'foreName', lastName: 'lastName'})
+      spyOn(pupilAccessArrangementsDataService, 'sqlDeletePupilsAccessArrangements')
+      const pupilData = await pupilAccessArrangementsService.deletePupilAccessArrangements('urlSlug', 9991001)
+      expect(pupilData).toEqual({id: 1, foreName: 'foreName', lastName: 'lastName'})
+      expect(pupilDataService.sqlFindOneBySlugAndSchool).toHaveBeenCalled()
+      expect(pupilAccessArrangementsDataService.sqlDeletePupilsAccessArrangements).toHaveBeenCalled()
+    })
+    it('rejects if url slug is not present', async () => {
+      spyOn(pupilDataService, 'sqlFindOneBySlugAndSchool')
+      spyOn(pupilAccessArrangementsDataService, 'sqlDeletePupilsAccessArrangements')
+      try {
+        await pupilAccessArrangementsService.deletePupilAccessArrangements()
```

Comment:
I think it would be prudent here to add in a `fail()` line after the `await` in case it doesn't throw.

Reply:
@jon-shipley had some issues with fail but apparently it works ok.
Updated.